simulation/vehicle/traffic_creator.py

```diff
@@ -0,0 +1,96 @@
+from common.config_reader import ConfigReader
+from simulation.vehicle.vehicle import Vehicle
+import random
+
+
+class TrafficCreator(object):
+    __percentages = {
+        "aggressive": ConfigReader.get_data("driving.traffic.driver_profile_type.aggressive"),
+        "moderate": ConfigReader.get_data("driving.traffic.driver_profile_type.moderate"),
+        "defensive": ConfigReader.get_data("driving.traffic.driver_profile_type.defensive")
+    }
+    __traffic = []
+    __instance = None
+
+    def __init__(self, map1):
+
+        if TrafficCreator.__instance is None:
+            TrafficCreator.__traffic = TrafficCreator.__create_traffic(map1)
+        else:
+            raise Exception("Object already exists")
+
+    @staticmethod
+    def get_instance(map1):
```

Comment:
Is this implementation related to some decision in the workflow? I mean to create an instance of the traffic creator.